docs/architecture.md

```diff
@@ -0,0 +1,292 @@
+# All About `armada-spark`
+
```

Comment:
Can you add a Table of Content at the beginning?

You can use [markdown-toc](https://github.com/jonschlinkert/markdown-toc)

Reply:
If necessary sure, but GitHub already automatically generates a TOC if viewed through Github.